source/routing/idx_map.h

```diff
@@ -33,16 +33,18 @@ www.navitia.io
 #include <vector>
 #include <iterator>
 #include <boost/range/iterator_range_core.hpp>
-#include "type/type.h" // for type::idx_t
 
 namespace navitia { namespace routing {
 
+typedef uint32_t idx_t;
+const idx_t invalid_idx = std::numeric_limits<idx_t>::max();
```

Comment:
The problem is that idx_map.h depends on type.h if we don't do that. I can move idx_map.h in utils, and define the idx_t of type.h as a typedef of the one of idx_map.h, if it's ok for you.